test/check-application

```diff
@@ -27,37 +27,111 @@ class TestApplication(testlib.MachineCase):
         b.click("#containers-containers-filter button")
-        b.wait_visible("#containers-containers-filter .dropdown-menu")
+        b.wait_present("#containers-containers-filter .dropdown-menu")
         b.click("#containers-containers-filter li[data-data='all'] a")
         b.wait_present("#containers-containers")
         b.wait_present('#containers-containers tr:contains("busybox:latest")')
-        b.wait_visible('#containers-containers tr:contains("busybox:latest")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest")')
+        # check busybox is running and alpine is exited
+        b.wait_present('#containers-containers tr:contains("busybox:latest") td:contains("running")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest") td:contains("exited")')
 
         # show container listing toggle
         b.click('#containers-containers tbody tr:contains("busybox:latest") td.listing-ct-toggle')
         b.wait_present('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
+        # test the container stop dropdown menu under the listing toggle
+        self.containerStopDropdown()
 
+        #show running container
         b.click("#containers-containers-filter button")
         b.wait_present("#containers-containers-filter .dropdown-menu")
         b.click("#containers-containers-filter li[data-data='running'] a")
+        b.wait_present('#containers-containers tr:contains("busybox:latest")')
+        b.wait_present('#containers-containers tr:contains("swamped-crate")')
+        b.wait_present('#containers-containers tr:contains("busybox:latest") + tr:contains("sleep 1000")')
+
+        # switch to all containers
+        b.wait_present("#containers-containers-filter button")
+        b.click("#containers-containers-filter button")
+        b.wait_present("#containers-containers-filter .dropdown-menu")
+        b.click("#containers-containers-filter li[data-data='all'] a")
         b.wait_present("#containers-containers")
-        b.wait_in_text("#containers-containers", "No running containers")
+        b.wait_present('#containers-containers tr:contains("busybox:latest") td:contains("running")')
+        b.wait_present('#containers-containers tr:contains("test-sh")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest") td:contains("exited")')
+
+        # delete a stopped container
+        b.click('#containers-containers tbody tr:contains("alpine:latest") td.listing-ct-toggle')
+        b.wait_present('#containers-containers tbody tr:contains("alpine:latest") + tr button.btn-delete')
+        b.click('#containers-containers tbody tr:contains("alpine:latest") + tr button.btn-delete')
+        b.wait_present(".modal-dialog div")
+        b.wait_present(".modal-dialog div .btn-danger")
+        b.click(".modal-dialog div .btn-danger")
+        b.wait_not_present(".modal-dialog div")
+        self.pageReload()
```

Comment:
If this is required, it hides a bug. Operations that are done through the UI should never require a page reload, only operations that are done on the command line or elsewhere. If this is a necessary workaround for now, it needs a "FIXME" comment with a detailled explanation.